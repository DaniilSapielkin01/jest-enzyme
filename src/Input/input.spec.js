import React from "react";
import Input from "./input";

describe("Input component", () => {
  it("Should render input", () => {
    const component = render(<Input />);
    expect(component).toMatchSnapshot();
  });

  describe("defaultProps", () => {
    it("value", () => {
      const component = shallow(<Input value={"Test"} />);
      expect(component).toMatchSnapshot();
    });

    it("handle - onChange", () => {
      const result = Input.defaultProps.onChange();
      expect(result).toBe(undefined);
    });
    it("handle - onKeyPress", () => {
      const result = Input.defaultProps.onKeyPress();
      expect(result).toBe(undefined);
    });
  });
});
