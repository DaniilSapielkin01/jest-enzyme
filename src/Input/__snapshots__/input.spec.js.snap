// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render input 1`] = `
<div
  class="inputWrapper"
>
  <i
    class="fas fa-search"
  />
  <input
    class="input"
    placeholder="Click to search"
    value=""
  />
</div>
`;

exports[`Input component defaultProps value 1`] = `
<div
  class="inputWrapper"
>
  <i
    class="fas fa-search"
  />
  <input
    class="input"
    placeholder="Click to search"
    value="Test"
  />
</div>
`;
